src/templates/results/components/GraphLayout.js

```diff
@@ -11,75 +13,154 @@ const GraphLayout = (props) => {
 
     const targetTaxon = results['Input'][0];
 
-    const graphData = {
-        nodes: [{
+    const [interactionTypes, setInteractionTypes] = useState();
+    const [nodeTracker, setNodeTracker] = useState({
+        [targetTaxon['taxon_id']]: {
             id: targetTaxon['taxon_id'],
             name: targetTaxon['sci_name'],
-            color: '#4b6c2d'
-        }],
-        links: []
-    };
-
-    const keyNodeTracker = [targetTaxon['taxon_id']];
-
-    /* For each observed taxon, add to graph data */
-    if (results['Predicted'].length > 0) {
-        results['Predicted'].forEach((taxon, _i) => {
-            graphData['nodes'].push({
-                id: taxon['taxon_id'],
-                name: taxon['sci_name']
+            color: '#4b6c2d',
+            interaction: formData['interaction']
+        }
+    });
+    const [linkTracker, setLinkTracker] = useState({});
+    const [keyNodeTracker, setKeyNodeTracker] = useState([targetTaxon['taxon_id']]);
+
+    useEffect(() => {
+        GetInteractions(Process);
+
+        function Process(result) {
+            setInteractionTypes(result);
+        }
+    }, []);
+
+    useEffect(() => {
+        /* For each observed taxon, add to graph data */
```

Comment:
Do you want to impose a limit here? Too many observed interactions might be overcrowded. Maybe the user can set the limit?

Reply:
This is something for discussion, we could set it in the code itself or let the user somehow do it